Análise exploratória dados fitbit

In [2]:
import pandas as pd

In [3]:
d_activity = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/dailyActivity_merged.csv')
heartrate = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/heartrate_seconds_merged.csv')
h_calories = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/hourlyCalories_merged.csv')
h_intensity = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/hourlyIntensities_merged.csv')
m_calories = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/minuteCaloriesNarrow_merged.csv')
m_intensity = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/minuteIntensitiesNarrow_merged.csv')
m_sleep = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/minuteSleep_merged.csv')
weightlog = pd.read_csv('D:/python/bellabeat-analysis/database/fitbit-data/weightLogInfo_merged.csv')

### Análise exploratória d_activity


In [4]:
d_activity.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 457 entries, 0 to 456
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        457 non-null    int64  
 1   ActivityDate              457 non-null    object 
 2   TotalSteps                457 non-null    int64  
 3   TotalDistance             457 non-null    float64
 4   TrackerDistance           457 non-null    float64
 5   LoggedActivitiesDistance  457 non-null    float64
 6   VeryActiveDistance        457 non-null    float64
 7   ModeratelyActiveDistance  457 non-null    float64
 8   LightActiveDistance       457 non-null    float64
 9   SedentaryActiveDistance   457 non-null    float64
 10  VeryActiveMinutes         457 non-null    int64  
 11  FairlyActiveMinutes       457 non-null    int64  
 12  LightlyActiveMinutes      457 non-null    int64  
 13  SedentaryMinutes          457 non-null    int64  
 14  Calories  

In [ ]:
d_activity['Id'].nunique() #35 ids nos dados de d_activity

35

### Análise Total steps
* os outros 34 Ids tem input de passos, sem dados discrepantes preocupantes
* um Id não inputou passos nunhuma vez
* vários ids (14) tem dias com 0 passos, como não é viável passar 1 dia sem dar passos, pode-se supor que são pessoas que não usam o fit bit em dias sem exercício

In [35]:
d_activity['TotalSteps'].median() #5986.0
d_activity['TotalSteps'].mean() #6546.5
d_activity['TotalSteps'].mode() #0  61 linhas desse dataframe não tem nenhuma entrada de exercício, ou passos. acho que daria para contar como dados nulos (pessoa não usou o fitbit)
#396 entradas com input de passos

0    0
Name: TotalSteps, dtype: int64

In [ ]:
pd.pivot_table(d_activity, values = 'TotalSteps', index= 'Id', aggfunc= ('mean', 'max')).round(2)
#apenas uma pessoa não fez input de passos, média e máximo de passos parece sem erros

max      mean
Id                         
1503960366  17609  11640.53
1624580081  18464   4226.26
1644430081  20237   9274.80
1844505072   6847   3640.58
1927972279   5662   2180.83
2022484408  18247  12174.92
2026352035   6679   3392.75
2320127002  11230   3138.42
2347167796  16081   9800.07
2873212765  14103   6636.58
2891001357   4126    773.62
3372868164   8827   6127.50
3977333714  15425   8663.92
4020332650  12483   5776.59
4057192912   8433   1887.22
4319703577  10946   7820.58
4388161847      0      0.00
4445114986   6911   4293.07
4558609924   8828   5785.25
4702921684  14002   7943.40
5553957443  15444   8354.75
5577150313  14873   8608.18
6117666160  19658   8248.60
6290855005   9548   1618.10
6391747486   9766   1336.89
6775888955  13987   5559.00
6962181067  18952  12639.71
7007744171  16806  12260.33
7086361926  14604   6103.92
8053475328  25701  14844.36
8253242879  12026   2389.92
8378563200  14179   8135.25
8583815059   6835   3045.50
8792009665   8452   3094.92
8877689391  28497  17417.08

In [ ]:
d_activity[d_activity['TotalSteps']== 0] #dados nulos? é possível entender que essas pessoas só usam o fitbit em dias de exercício?

51     1844505072
52     1844505072
59     1844505072
96     2320127002
97     2320127002
          ...    
402    8253242879
403    8253242879
404    8253242879
405    8253242879
412    8253242879
Name: Id, Length: 61, dtype: int64

In [ ]:
# pesquisando entradas dos ids com steps nulos - 14 pessoas com dados nulos de passos - sabendo que apenas um desses Ids (4388161847) não tem nunhum input de passos
null_steps_ids = d_activity.loc[d_activity['TotalSteps']== 0, 'Id'].drop_duplicates()
null_steps_ids

51     1844505072
96     2320127002
122    2347167796
134    2873212765
135    2891001357
197    4057192912
240    4319703577
241    4388161847
290    4702921684
316    6117666160
326    6290855005
334    6391747486
346    6775888955
401    8253242879
Name: Id, dtype: int64

### Análise exploratória heartrate
* apenas 14 ids tem input de heart rate
* 4 ids em comum com a lista de pessoas que teorizei que só usam durante exercício (6775888955 , 6391747486, 6117666160 , 2347167796), o que ainda pode fazer sentido ,a valor max de batimento é alto


In [72]:
heartrate.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1154681 entries, 0 to 1154680
Data columns (total 3 columns):
 #   Column  Non-Null Count    Dtype 
---  ------  --------------    ----- 
 0   Id      1154681 non-null  int64 
 1   Time    1154681 non-null  object
 2   Value   1154681 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 26.4+ MB


In [ ]:
heartrate['Id'].nunique() #14 ids com heartrate, menos que a metade dos dados de passos

14

In [ ]:
pd.pivot_table(heartrate, values = 'Value', index= 'Id', aggfunc= ['mean', 'max', 'min']).round(2).sort_values(by= ('mean', 'Value'), ascending= False)

mean   max   min
            Value Value Value
Id                           
6775888955  97.74   185    59
7007744171  89.93   158    56
8877689391  87.60   179    47
6391747486  84.12   133    49
6117666160  83.49   130    52
4020332650  81.90   151    46
2022484408  81.69   182    46
6962181067  80.69   176    48
4558609924  79.55   158    50
8792009665  76.33   163    45
2347167796  76.07   182    51
5553957443  69.42   152    48
2026352035  65.78    80    57
5577150313  65.00   176    36

### Análise exploratória hourly calories
* 34 ids tem input de calorias
continuar daqui

In [88]:
h_calories.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24084 entries, 0 to 24083
Data columns (total 3 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Id            24084 non-null  int64 
 1   ActivityHour  24084 non-null  object
 2   Calories      24084 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 564.6+ KB


In [ ]:
h_calories['Id'].nunique() #34 ids com input de calorias (gastas)

34

In [ ]:
h_calories['ActivityDay'] = h_calories['ActivityHour'].str.extract(r'^([^\s]+)')

pd.pivot_table(h_calories, values= 'Calories', index= ('Id', 'ActivityDay'), aggfunc= 'sum')


#quero agora fazer a média de gasto calórico de cada id (usando o gasto calórico dos dias), para que cada id tenha uma linha com 'average calories', mas não sei como fazer, preciso pesquisar


Calories
Id         ActivityDay          
1503960366 3/12/2016        2228
           3/13/2016        2100
           3/14/2016        1830
           3/15/2016        2111
           3/16/2016        1967
...                          ...
8877689391 4/5/2016         2832
           4/6/2016         4034
           4/7/2016         2941
           4/8/2016         4191
           4/9/2016         3835

[1021 rows x 1 columns]

### Análise de sobrepeso nos usários do fitbit
* 35 ids tem dados sobre peso
* 7 ids do fit estão acima do peso, e apenas 1 delas tem obesidade grau 2 e apenas 1 tem obesidade grau 3
* comparando weightlog e daily aactivity, 3 usuários acima do peso  tem inputs com total step zerado ,mas não todos os inputs de total steps zerados

In [31]:
d_activity[d_activity['Id'].isin(overweight_ids)]
# 35 unique IDs, 457 entradas de dados

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
60   1927972279     4/1/2016        4317       2.990000         2.990000   
61   1927972279     4/2/2016        5662       3.920000         3.920000   
62   1927972279     4/3/2016        3198       2.210000         2.210000   
63   1927972279     4/4/2016        2352       1.630000         1.630000   
64   1927972279     4/5/2016        2234       1.550000         1.550000   
..          ...          ...         ...            ...              ...   
452  8877689391     4/8/2016       23014      20.389999        20.389999   
453  8877689391     4/9/2016       16470       8.070000         8.070000   
454  8877689391    4/10/2016       28497      27.530001        27.530001   
455  8877689391    4/11/2016       10622       8.060000         8.060000   
456  8877689391    4/12/2016        2350       1.780000         1.780000   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
60                        0.0                0.00                      0.37   
61                        0.0                0.00                      0.00   
62                        0.0                0.00                      0.00   
63                        0.0                0.00                      0.00   
64                        0.0                0.00                      0.00   
..                        ...                 ...                       ...   
452                       0.0               11.10                      0.63   
453                       0.0                0.00                      0.02   
454                       0.0               21.92                      1.12   
455                       0.0                1.47                      0.15   
456                       0.0                0.00                      0.00   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
60                  2.62                     0.00                  0   
61                  3.92                     0.00                  0   
62                  2.21                     0.00                  0   
63                  1.63                     0.00                  0   
64                  1.55                     0.00                  0   
..                   ...                      ...                ...   
452                 8.62                     0.00                 70   
453                 8.02                     0.00                 90   
454                 4.46                     0.00                128   
455                 6.37                     0.01                 18   
456                 1.78                     0.00                  0   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
60                    11                   192               854      2590  
61                     0                   267               858      2783  
62                     0                   146              1183      2449  
63                     0                   128               829      2380  
64                     0                   108               912      2344  
..                   ...                   ...               ...       ...  
452                   29                   359               982      4196  
453                    9                   289              1052      3841  
454                   46                   211              1055      4526  
455                    7                   225              1190      2820  
456                    0                    58               531       938  

[86 rows x 15 columns]

In [18]:
weightlog[weightlog['BMI']> 25].sort_values('BMI', ascending= False )

Id                   Date    WeightKg  WeightPounds  Fat  \
1   1927972279   4/10/2016 6:33:26 PM  129.600006    285.719105  NaN   
6   4445114986  3/30/2016 11:59:59 PM   92.400002    203.707134  NaN   
23  8253242879   4/7/2016 11:59:59 PM   75.599998    166.669467  NaN   
7   4558609924   4/8/2016 11:59:59 PM   69.400002    153.000813  NaN   
8   4702921684   4/4/2016 11:59:59 PM   99.699997    219.800869  NaN   
25  8877689391    4/4/2016 6:53:43 AM   86.599998    190.920316  NaN   
27  8877689391    4/6/2016 6:49:31 AM   86.300003    190.258939  NaN   
31  8877689391   4/11/2016 6:58:09 AM   86.099998    189.818004  NaN   
26  8877689391    4/5/2016 6:40:15 AM   86.000000    189.597545  NaN   
32  8877689391   4/12/2016 6:47:11 AM   85.800003    189.156628  NaN   
24  8877689391    4/1/2016 6:49:40 AM   85.500000    188.495234  NaN   
30  8877689391    4/9/2016 8:06:12 AM   85.400002    188.274775  NaN   
28  8877689391    4/7/2016 6:15:08 AM   85.099998    187.613382  NaN   
29  8877689391    4/8/2016 6:39:44 AM   85.000000    187.392923  NaN   
5   2891001357   4/5/2016 11:59:59 PM   88.400002    194.888643  NaN   

          BMI  IsManualReport          LogId  
1   46.169998           False  1460313206000  
6   35.009998            True  1459382399000  
23  29.549999            True  1460073599000  
7   27.139999            True  1460159999000  
8   26.110001            True  1459814399000  
25  25.940001           False  1459752823000  
27  25.830000           False  1459925371000  
31  25.790001           False  1460357889000  
26  25.760000           False  1459838415000  
32  25.680000           False  1460443631000  
24  25.610001           False  1459493380000  
30  25.559999           False  1460189172000  
28  25.490000           False  1460009708000  
29  25.440001           False  1460097584000  
5   25.030001            True  1459900799000

In [29]:
overweight_ids = weightlog.loc[weightlog['BMI'] > 25, 'Id'].unique()

In [30]:
weightlog[weightlog['Id'].isin(overweight_ids)]

Id                   Date    WeightKg  WeightPounds  Fat  \
1   1927972279   4/10/2016 6:33:26 PM  129.600006    285.719105  NaN   
5   2891001357   4/5/2016 11:59:59 PM   88.400002    194.888643  NaN   
6   4445114986  3/30/2016 11:59:59 PM   92.400002    203.707134  NaN   
7   4558609924   4/8/2016 11:59:59 PM   69.400002    153.000813  NaN   
8   4702921684   4/4/2016 11:59:59 PM   99.699997    219.800869  NaN   
23  8253242879   4/7/2016 11:59:59 PM   75.599998    166.669467  NaN   
24  8877689391    4/1/2016 6:49:40 AM   85.500000    188.495234  NaN   
25  8877689391    4/4/2016 6:53:43 AM   86.599998    190.920316  NaN   
26  8877689391    4/5/2016 6:40:15 AM   86.000000    189.597545  NaN   
27  8877689391    4/6/2016 6:49:31 AM   86.300003    190.258939  NaN   
28  8877689391    4/7/2016 6:15:08 AM   85.099998    187.613382  NaN   
29  8877689391    4/8/2016 6:39:44 AM   85.000000    187.392923  NaN   
30  8877689391    4/9/2016 8:06:12 AM   85.400002    188.274775  NaN   
31  8877689391   4/11/2016 6:58:09 AM   86.099998    189.818004  NaN   
32  8877689391   4/12/2016 6:47:11 AM   85.800003    189.156628  NaN   

          BMI  IsManualReport          LogId  
1   46.169998           False  1460313206000  
5   25.030001            True  1459900799000  
6   35.009998            True  1459382399000  
7   27.139999            True  1460159999000  
8   26.110001            True  1459814399000  
23  29.549999            True  1460073599000  
24  25.610001           False  1459493380000  
25  25.940001           False  1459752823000  
26  25.760000           False  1459838415000  
27  25.830000           False  1459925371000  
28  25.490000           False  1460009708000  
29  25.440001           False  1460097584000  
30  25.559999           False  1460189172000  
31  25.790001           False  1460357889000  
32  25.680000           False  1460443631000

Apenas 7 id únicos pessoas dos dados usuárias do fit estão acima do peso, e apenas 1 delas tem obesidade grau 2 e apenas 1 tem obesidade grau 3

### Analisar os dados desse grupo amostral nos outros csv's

In [108]:
d_activity[(d_activity['TotalSteps']== 0) & (d_activity['Id'].isin(overweight_ids))] #3 usuários acima do peso  tem inputs com total step zerado

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
135  2891001357    3/29/2016           0            0.0              0.0   
136  2891001357    3/30/2016           0            0.0              0.0   
137  2891001357    3/31/2016           0            0.0              0.0   
138  2891001357     4/1/2016           0            0.0              0.0   
140  2891001357     4/3/2016           0            0.0              0.0   
141  2891001357     4/4/2016           0            0.0              0.0   
290  4702921684    4/12/2016           0            0.0              0.0   
401  8253242879     4/1/2016           0            0.0              0.0   
402  8253242879     4/2/2016           0            0.0              0.0   
403  8253242879     4/3/2016           0            0.0              0.0   
404  8253242879     4/4/2016           0            0.0              0.0   
405  8253242879     4/5/2016           0            0.0              0.0   
412  8253242879    4/12/2016           0            0.0              0.0   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
135                  0.000000                 0.0                       0.0   
136                  0.000000                 0.0                       0.0   
137                  0.000000                 0.0                       0.0   
138                  4.828032                 0.0                       0.0   
140                  0.000000                 0.0                       0.0   
141                  0.000000                 0.0                       0.0   
290                  0.000000                 0.0                       0.0   
401                  0.000000                 0.0                       0.0   
402                  0.000000                 0.0                       0.0   
403                  0.000000                 0.0                       0.0   
404                  0.000000                 0.0                       0.0   
405                  0.000000                 0.0                       0.0   
412                  0.000000                 0.0                       0.0   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
135                  0.0                      0.0                  0   
136                  0.0                      0.0                  0   
137                  0.0                      0.0                  0   
138                  0.0                      0.0                  0   
140                  0.0                      0.0                  0   
141                  0.0                      0.0                  0   
290                  0.0                      0.0                  0   
401                  0.0                      0.0                  0   
402                  0.0                      0.0                  0   
403                  0.0                      0.0                  0   
404                  0.0                      0.0                  0   
405                  0.0                      0.0                  0   
412                  0.0                      0.0                  0   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
135                    0                     0              1440      1920  
136                    0                     0              1440      1920  
137                    0                     0              1440      1920  
138                  660                     0               780      4562  
140                    0                     0              1440      1920  
141                    0                     0              1440      1920  
290                    0                     0              1440         0  
401                    0                     0              1440      1429  
402                    0                     0              1440      1429  
403                    0                     0              14

# insights até aqui